### PR TITLE
[WIP] Simple Memory Checks

### PR DESCRIPTION
**Pull request by kuhar**
Work in progress

### CODE REVIEW

**Comment by kuhar** [comment]
Replaced plain dbgs() with LOG.

Also added instrumentation for GlobalVariables (both when determined potentially safe and unsafe). This appears to work as expected for when a GV is considered uninteresting (== safe), but not when we start tracking it (as in `test/smc/list1_global_unsat3.c`). For some reason, the BAR and FOO tags don't appear in shadow mem, things don't alias when they get to the searhon solver, and sea-dsa reports at most 1 alloc site for each pointer, and it doesn't want to produce a mem-dot file...

I will try to debug it some more.

File: lib/Transforms/Scalar/LowerGvInitializers.cc [hunk 1]
```diff
@@ -1,240 +1,242 @@
 #include "seahorn/Transforms/Scalar/LowerGvInitializers.hh"
 
-#include "boost/range.hpp"
 #include "boost/format.hpp"
+#include "boost/range.hpp"
```
Comment (kuhar) [inline lib/Transforms/Scalar/LowerGvInitializers.cc]: I should have probably split formatting changes and the fix to lowering. Is is better to leave it as-is, or make into 2 separate commits (the formatting one in a separate pull request)?

Comment (agurfinkel) [reply lib/Transforms/Scalar/LowerGvInitializers.cc]: 2 separate commits so that it is clear what is your change and what is formatting would be nice

**Comment by kuhar** [comment]
Closing for now. I created a new PR with fewer commits and a better description.

**Comment by agurfinkel** [comment]
@kuhar , the PR is not actually closed. Should I close it?

**Comment by kuhar** [comment]
Yes, seems like I cannot close it with the current permissions
